npm/src/commands/runTests.js

```diff
@@ -250,10 +252,27 @@ async function runTestGroup(
     // For each compatible version of Flow, run `flow check` and verify there
     // are no errors.
     const flowVersionsToRun = orderedFlowVersions.filter(flowVer => {
-      return (
-        testGroup.flowVersion === "all"
-        || semver.satisfies(flowVer, testGroup.flowVersion)
-      );
+      if (testGroup.flowVersion === "all") {
+        return true;
+      }
+
+      const testGrpUpperBound = testGroup.flowVersion.upperBound;
+      testGroup.flowVersion.upperBound = undefined;
```

Comment:
If there's an upperbound, `versionToString()` would render a string like "v0.1.0_<=v0.2.0", but we want the two separate so we can use the `semver` module. So to handle this I'm just temporarily separating the two versions for string-rendering reasons.

I could also split on `'_'`, but it would just be a one-off place where we define the schema for how upper/lower bounds are formatted with each other (i.e. if we change this again we'd have to remember to change it here and in the utils modules)